extension/test/test-logger.js

```diff
@@ -27,7 +33,12 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
     }
     finally {
       istream.close();
-      message = JSON.parse(message);
+      try {
+        message = JSON.parse(message);
+      } catch (e) {
+        Cu.reportError("Invalid JSON string: " + message);
+        throw e;
+      }
```

Comment:
I don't think there's a doesntThrow(). What's stopping you from just using console.log() to log whatever you want to the console?